src/java.base/share/classes/java/io/ClassCache.java

```diff
@@ -39,24 +40,20 @@
 
     private static class CacheRef<T> extends SoftReference<T> {
         private final Class<?> type;
-        private T strongReferent;
+        private final AtomicReference<T> strongReferent;
```

Comment:
This introduces a dereference. Maybe it should be `AtomicReferenceFieldUpdater` or a `VarHandle`, assuming there are no bootstrapping issues with this code.